src/event-progress-bar.js

```diff
@@ -15,49 +15,58 @@ function doProgress() {
 
 	// Bail if there are no live events.
 	if ( events.length <= 0 ) {
-		//clearInterval( x );
 		return;
 	}
 
 	// Get today's date and time and the offset from UTC.
 	var d = new Date();
 	var now = d.getTime();
 	var tzOffset = d.getTimezoneOffset();
-	var offset = tzOffset * 60 * 1000;
+	var offset = tzOffset * 1000 * 60;
 
 	// Go through all the events that are running at the moment.
-	for( let i = 0; i < events.length; i++ ) {
-		var startDate = new Date( events[ i ].children[ 0 ].dataset.startDate ).getTime();
-		var endDate = new Date( events[ i ].children[ 0 ].dataset.endDate ).getTime();
+	Array.prototype.forEach.call( events, function( event ) {
+		var startDate = new Date( event.querySelector( '.progress-bar-container__live' ).dataset.startDate ).getTime();
+		var endDate = new Date( event.querySelector( '.progress-bar-container__live' ).dataset.endDate ).getTime();
 
 		var fullTime = endDate - startDate;
 		var timeLeft = endDate - (now + offset);
 		var percent = (fullTime - timeLeft) / fullTime * 100;
 
-		// If the count down is over, write some text and get out of the loop.
 		if ( timeLeft < 0 ) {
-			events[ i ].children[ 0 ].innerHTML = "Event";
-			events[ i ].children[ 2 ].children[ 0 ].innerHTML = "is over.";
-			events[ i ].classList.remove( "progress-bar-container__on" );
-			continue;
-		}
+			// If the countdown is over change the labels.
+			event.querySelector( '.progress-bar-container__live-text' ).classList.add( 'tribe-common-a11y-hidden' );
+			event.querySelector( '.progress-bar-container__live-text--over' ).classList.remove( 'tribe-common-a11y-hidden' );
+			event.querySelector( '.progress-bar-container__timeleft-time' ).classList.add( 'tribe-common-a11y-hidden' );
+			event.querySelector( '.progress-bar-container__timeleft-string' ).classList.add( 'tribe-common-a11y-hidden' );
+			event.querySelector( '.progress-bar-container__timeleft--over' ).classList.remove( 'tribe-common-a11y-hidden' );
+			event.classList.remove( "progress-bar-container__on" );
+		} else {
+			// Time calculations for days, hours, minutes and seconds
+			var second = 1000;
+			var minute = second * 60; // (1000 * 60)
+			var hour = minute * 60;   // (1000 * 60 * 60)
+			var day = hour * 24;      // (1000 * 60 * 60 * 24)
```

Comment:
In the let/const world - these would be const - they don't change.

So it occured to me we could move these up out of the foreach (up where we define `tzOffset`, etc) - then we're not redefining them each iteration of the loop.

Reply:
That makes so much sense!